src/config.rs

```diff
@@ -210,17 +234,14 @@ impl Config {
             self.default_channels = other.default_channels.clone();
         }
 
-        if other.change_ps1.is_some() {
-            self.change_ps1 = other.change_ps1;
-        }
-
-        if other.tls_no_verify.is_some() {
-            self.tls_no_verify = other.tls_no_verify;
-        }
-
-        if other.authentication_override_file.is_some() {
-            self.authentication_override_file = other.authentication_override_file.clone();
-        }
+        assign_if_some!(
+            self,
+            other,
+            change_ps1,
+            tls_no_verify,
+            authentication_override_file,
+            repodata_config
+        );
```

Comment:
I dont think this justifies a macro. Its much harder to parse. Although there is some copy-paste risk I think this is much cleaner:

```suggestion
        self.change_ps1 = other.change_ps1.or(self.change_ps1);
        self.tls_no_verify= other.tls_no_verify.or(self.tls_no_verify);
```

I also think that you could write this nicer by doing:

```rust
self = Self {
   change_ps1: other.change_ps1.or(self.change_ps1),
   ...
}
```

That way you are automatically notified by the compiler that you should update this if a field is added.

Finally, instead of taking `&mut self` it might be nicer to take `mut self` and return `Self`.

Reply:
Done